orderer/consensus/smartbft/consenter.go

```diff
@@ -0,0 +1,271 @@
+	"github.com/hyperledger/fabric/common/policies"
+	"github.com/hyperledger/fabric/internal/pkg/comm"
+	"github.com/hyperledger/fabric/orderer/common/cluster"
+	"github.com/hyperledger/fabric/orderer/common/localconfig"
+	"github.com/hyperledger/fabric/orderer/common/multichannel"
+	"github.com/hyperledger/fabric/orderer/consensus"
+	"github.com/hyperledger/fabric/protoutil"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+)
+
+// CreateChainCallback creates a new chain
+type CreateChainCallback func()
+
+// ChainGetter obtains instances of ChainSupport for the given channel
+type ChainGetter interface {
+	// GetChain obtains the ChainSupport for the given channel.
+	// Returns nil, false when the ChainSupport for the given channel
+	// isn't found.
+	GetChain(chainID string) *multichannel.ChainSupport
+}
+
+// PolicyManagerRetriever is the policy manager retriever function
+type PolicyManagerRetriever func(channel string) policies.Manager
+
+// Consenter implementation of the BFT smart based consenter
+type Consenter struct {
+	CreateChain      func(chainName string)
+	GetPolicyManager PolicyManagerRetriever
+	Logger           *flogging.FabricLogger
+	Cert             []byte
+	Comm             *cluster.Comm
+	Chains           ChainGetter
+	SignerSerializer SignerSerializer
+	Registrar        *multichannel.Registrar
+	WALBaseDir       string
+	ClusterDialer    *cluster.PredicateDialer
+	Conf             *localconfig.TopLevel
+	Metrics          *Metrics
+	BCCSP            bccsp.BCCSP
+}
+
+// New creates Consenter of type smart bft
+func New(
+	pmr PolicyManagerRetriever,
+	signerSerializer SignerSerializer,
+	clusterDialer *cluster.PredicateDialer,
+	conf *localconfig.TopLevel,
+	srvConf comm.ServerConfig,
+	srv *comm.GRPCServer,
+	r *multichannel.Registrar,
+	metricsProvider metrics.Provider,
+	BCCSP bccsp.BCCSP,
+) *Consenter {
+	logger := flogging.MustGetLogger("orderer.consensus.smartbft")
+
+	metrics := cluster.NewMetrics(metricsProvider)
+
+	var walConfig WALConfig
+	err := mapstructure.Decode(conf.Consensus, &walConfig)
+	if err != nil {
+		logger.Panicf("Failed to decode consensus configuration: %s", err)
+	}
+
+	logger.Infof("WAL Directory is %s", walConfig.WALDir)
+
+	consenter := &Consenter{
+		Registrar:        r,
+		GetPolicyManager: pmr,
+		Conf:             conf,
+		ClusterDialer:    clusterDialer,
+		Logger:           logger,
+		Cert:             srvConf.SecOpts.Certificate,
+		Chains:           r,
+		SignerSerializer: signerSerializer,
+		WALBaseDir:       walConfig.WALDir,
+		Metrics:          NewMetrics(metricsProvider),
+		CreateChain:      r.CreateChain,
+		BCCSP:            BCCSP,
+	}
+
+	compareCert := cluster.CachePublicKeyComparisons(func(a, b []byte) bool {
+		err := crypto.CertificatesWithSamePublicKey(a, b)
+		if err != nil && err != crypto.ErrPubKeyMismatch {
+			crypto.LogNonPubKeyMismatchErr(logger.Errorf, err, a, b)
+		}
+		return err == nil
+	})
+
+	consenter.Comm = &cluster.Comm{
```

Comment:
why are we initializing this with `Comm`?

Aren't we supposed to use the new cluster service?

Reply:
Updated to use new commservice